<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>1&nbsp; Genetic Variant Data – A2CPS Starter Kits</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./genomic_imputation.html" rel="next">
<link href="./index.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-2fef5ea3f8957b3e4ecc936fc74692ca.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-5fd9db47a040b21ac2cac9a0b3b722ba.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-1a431d08dc38a67a43972170746f6e10.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="site_libs/bootstrap/bootstrap-5fd9db47a040b21ac2cac9a0b3b722ba.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-K3383Z78V7"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-K3383Z78V7', { 'anonymize_ip': true});
</script>


</head>

<body class="nav-sidebar floating quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./genetic_variants.html"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Genetic Variant Data</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">A2CPS Starter Kits</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/a2cps/starterkits" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./genetic_variants.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Genetic Variant Data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./genomic_imputation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Imputed Genomics</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./raw-mri.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Raw MRI Data Starter Kit</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./freesurfer.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">FreeSurfer Measures</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./brainager.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Brain Age</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cat12.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Voxel-Based Morphometry</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./fcn.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Functional Connectivity</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./gift.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Group ICA of fMRI Toolbox (GIFT)</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./neural-pain-signatures.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Prior Neural Pain Signature Responses</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./faq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./bids-qc-joining.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Exclude Scans with “Red” Quality</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ids.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">B</span>&nbsp; <span class="chapter-title">ID Information</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./nda-mri.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">C</span>&nbsp; <span class="chapter-title">NDA Download Example, MRI</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./psycho_social_biomarker_impute.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">D</span>&nbsp; <span class="chapter-title">Psychosocial Data Imputation and Biomarker calculations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./biospecimenCRF_quality_checks_and_reformat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">E</span>&nbsp; <span class="chapter-title">Biospecimen Collection CRF Data Quality Checks</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FuntionalCRF_data_quality_checks_and_reformat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">F</span>&nbsp; <span class="chapter-title">Functional CRF Data Quality Checks</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ImagingCRF_data_quality_checks_and_reformat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">G</span>&nbsp; <span class="chapter-title">Imaging CRF Data Quality Checks</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./QSTCRF_data_quality_checks_and_reformat.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">H</span>&nbsp; <span class="chapter-title">QST CRF Data Quality Checks</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#starting-project" id="toc-starting-project" class="nav-link active" data-scroll-target="#starting-project"><span class="header-section-number">1.1</span> Starting Project</a>
  <ul class="collapse">
  <li><a href="#locate-data" id="toc-locate-data" class="nav-link" data-scroll-target="#locate-data"><span class="header-section-number">1.1.1</span> Locate Data</a></li>
  </ul></li>
  <li><a href="#plink-binary-file-set" id="toc-plink-binary-file-set" class="nav-link" data-scroll-target="#plink-binary-file-set"><span class="header-section-number">1.2</span> Plink Binary File Set</a>
  <ul class="collapse">
  <li><a href="#extract-data" id="toc-extract-data" class="nav-link" data-scroll-target="#extract-data"><span class="header-section-number">1.2.1</span> Extract Data</a></li>
  </ul></li>
  <li><a href="#quality-control" id="toc-quality-control" class="nav-link" data-scroll-target="#quality-control"><span class="header-section-number">1.3</span> Quality Control</a>
  <ul class="collapse">
  <li><a href="#overall-filters" id="toc-overall-filters" class="nav-link" data-scroll-target="#overall-filters"><span class="header-section-number">1.3.1</span> Overall Filters</a></li>
  <li><a href="#considerations-while-working-with-these-data" id="toc-considerations-while-working-with-these-data" class="nav-link" data-scroll-target="#considerations-while-working-with-these-data"><span class="header-section-number">1.3.2</span> Considerations while working with these data</a></li>
  <li><a href="#quality-control-code" id="toc-quality-control-code" class="nav-link" data-scroll-target="#quality-control-code"><span class="header-section-number">1.3.3</span> Quality Control Code</a></li>
  <li><a href="#principal-component-analysis" id="toc-principal-component-analysis" class="nav-link" data-scroll-target="#principal-component-analysis"><span class="header-section-number">1.3.4</span> Principal Component Analysis</a></li>
  <li><a href="#sex-typing" id="toc-sex-typing" class="nav-link" data-scroll-target="#sex-typing"><span class="header-section-number">1.3.5</span> Sex Typing</a></li>
  <li><a href="#relatedness" id="toc-relatedness" class="nav-link" data-scroll-target="#relatedness"><span class="header-section-number">1.3.6</span> Relatedness</a></li>
  <li><a href="#imputation" id="toc-imputation" class="nav-link" data-scroll-target="#imputation"><span class="header-section-number">1.3.7</span> Imputation</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">


<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-snps" class="quarto-section-identifier"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Genetic Variant Data</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>This page provides considerations for A2CPS projects that involve the Genetic Variant data.</p>
<p><img src="https://upload.wikimedia.org/wikipedia/commons/f/ff/DNA_double_helix_%2813081113544%29.jpg" class="img-fluid" width="500"> <!-- This file is licensed under the Creative Commons Attribution 2.0 Generic license. --></p>
<p>Genetic variant data captures individual variation in the human genome with single-nucleotide resolution. Our released SNP array data includes 690,126 genetic loci across all 22 autosomes, the X / Y (sex) chromosomes, and mitochondrial DNA for 1375 participants. At each locus, there is a more common “reference” (REF) allele and a less common “alternate” (ALT) allele, also referred to as the major and minor alleles, respectively. Because humans are diploid, we carry two copies of each chromosome, and thus two alleles at each locus, which may be both REF, both ALT, or one of each (heterozygous). This data enables researchers to examine allele-level variation across the genome at precisely selected sites for each participant. Below, we expand with more granularity how to work with these data, and the quality control measures that were used to create them. Notably, we also used the quality controlled data to impute whole genome sequences using TOPMed’s Imputation Server, information that is included in the “Imputed Genomics” starter kit.</p>
<hr>
<section id="starting-project" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="starting-project"><span class="header-section-number">1.1</span> Starting Project</h2>
<section id="locate-data" class="level3" data-number="1.1.1">
<h3 data-number="1.1.1" class="anchored" data-anchor-id="locate-data"><span class="header-section-number">1.1.1</span> Locate Data</h3>
<p>On TACC, the data are stored underneath the releases. For example, data release <code>v2.1.0</code> is underneath</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/corral-secure/projects/A2CPS/products/consortium-data/pre-surgery-release-2-1-0/omics/gene_variants</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The genetic variant data are underneath the <code>omics/gene_variants</code> folder</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> ls /corral-secure/projects/A2CPS/products/consortium-data/pre-surgery-release-2-1-0/omics/gene_variants</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2.bed</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2.bim</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2.fam</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2_Imputed_Genomics.bed</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2_Imputed_Genomics.bim</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2_Imputed_Genomics.fam</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2_Imputed_Genomics.log</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2.log</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2.nosex</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="ex">2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2_PCA_Ind_Info.csv</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>These files are in the <a href="https://www.cog-genomics.org/plink/1.9/">PLINK</a> format.</p>
</section>
</section>
<section id="plink-binary-file-set" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="plink-binary-file-set"><span class="header-section-number">1.2</span> Plink Binary File Set</h2>
<section id="bim" class="level4" data-number="1.2.0.1">
<h4 data-number="1.2.0.1" class="anchored" data-anchor-id="bim"><span class="header-section-number">1.2.0.1</span> .bim</h4>
<p>Each row in this file specifics a genetic variant and provides extended information on its location and alleles.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> cd /corral-secure/projects/A2CPS/products/consortium-data/pre-surgery-release-2-1-0/omics/gene_variants</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> head <span class="at">-5</span> 2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2.bim</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell">
<div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 5%">
<col style="width: 23%">
<col style="width: 13%">
<col style="width: 13%">
<col style="width: 22%">
<col style="width: 22%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">CHR</th>
<th style="text-align: left;">VAR ID</th>
<th style="text-align: right;">POS (cM)</th>
<th style="text-align: right;">POS (bp)</th>
<th style="text-align: left;">ALLELE 1 (ALT)</th>
<th style="text-align: left;">ALLELE 2 (REF)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">rs9651229</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">632287</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
</tr>
<tr class="even">
<td style="text-align: right;">1</td>
<td style="text-align: left;">rs9701872</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">632828</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">T</td>
</tr>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">rs11497407</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">633147</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
</tr>
<tr class="even">
<td style="text-align: right;">1</td>
<td style="text-align: left;">GSA-rs116587930</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">792461</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
</tr>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">rs3131972</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">817341</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li>CHR = chromosome #</li>
<li>Var ID
<ul>
<li>rsID = Reference SNP (single nucleotide polymorphism) ID</li>
<li>GSA = Global Screening Array</li>
<li>22:24377823_CNV_GSTT1
<ul>
<li>For Variants labeled as such, this ID means CHR:POS:CopyNumberVariant:Gene</li>
</ul></li>
</ul></li>
<li>cM (centimorgan) = measure of the frequency genetic recombination. 1 cM is a 1% chance that two markers on a chr will separate due to recombination during meiosis. Often not included.</li>
<li>bp (base-pair) = The genomic position on a given chromosome.</li>
<li>Allele 1 = reference, major allele</li>
<li>Allele 2 = alternate, minor allele</li>
</ul>
</section>
<section id="fam" class="level4" data-number="1.2.0.2">
<h4 data-number="1.2.0.2" class="anchored" data-anchor-id="fam"><span class="header-section-number">1.2.0.2</span> .fam</h4>
<p>Each row in this file specifics a participant and provides extended information on its identifiers, family, and gender.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> head <span class="at">-5</span> 2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2.fam</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell">
<div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 13%">
<col style="width: 38%">
<col style="width: 13%">
<col style="width: 13%">
<col style="width: 5%">
<col style="width: 13%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">Family ID</th>
<th style="text-align: left;">Individual ID</th>
<th style="text-align: right;">Father ID</th>
<th style="text-align: right;">Mother ID</th>
<th style="text-align: right;">Sex</th>
<th style="text-align: right;">Phenotype</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">P3VO69LyyV2AZ_L27BDmybNy8zN</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">-9</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: left;">74n6ZMnXd3WmA_L27BDmybNy8zN</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">-9</td>
</tr>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: left;">Px3x6X3gWBNA1_L27BDmybNy8zN</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">-9</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: left;">eyPoDR1qoJegX_L27BDmybNy8zN</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">-9</td>
</tr>
<tr class="odd">
<td style="text-align: right;">5</td>
<td style="text-align: left;">74m9r0BKoMxk3_L27BDmybNy8zN</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">-9</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li>Family ID (FID) = identifier for the participant’s family</li>
<li>Individual ID (IID) = identifier for the participant, in the form biospecimenID_runlistID</li>
<li>Father ID (FaID) = identifier for the participant’s father. A2CPS not ascertained based on family structure.</li>
<li>Mother ID (MoID) = identifier for the participant’s mother. A2CPS not ascertained based on family structure.</li>
<li>Sex = 1 : M, 2 : F</li>
<li>Phenotype = 1 = Control, 2 = Case, 0/-9 if not present. Often not included.</li>
</ul>
</section>
<section id="bed" class="level4" data-number="1.2.0.3">
<h4 data-number="1.2.0.3" class="anchored" data-anchor-id="bed"><span class="header-section-number">1.2.0.3</span> .bed</h4>
<p>A binary file, unviewable through command line commands such as “cat”. This contains each participant’s genotype data. Stored as bytes:</p>
<p>00 Homozygous for first allele in .bim file</p>
<p>01 Missing genotype</p>
<p>10 Heterozygous</p>
<p>11 Homozygous for second allele in .bim file</p>
<p>To see all files types in plink, go here: <a href="https://www.cog-genomics.org/plink/1.9/formats">File format reference - PLINK 1.9</a>.</p>
<hr>
</section>
<section id="extract-data" class="level3" data-number="1.2.1">
<h3 data-number="1.2.1" class="anchored" data-anchor-id="extract-data"><span class="header-section-number">1.2.1</span> Extract Data</h3>
<p>To access the variant data, plink2 (or plink 1.9) must be installed. Please see: <a href="https://www.cog-genomics.org/plink/2.0/">PLINK 2.0</a> or <a href="https://www.cog-genomics.org/plink/1.9/">PLINK 1.9</a>.</p>
<p>Here, we are storing a set of genetic variants from our primary biomarker list in a text file called ‘snps.txt’. Since we specifically probed these sites, they have longer, study-specific identifiers. By contrast, a typical SNP is assigned a reference SNP ID (rsID), which usually takes the form rs followed by a number, such as rs1, rs105, or rs123456789.</p>
<pre class="shell"><code>$ cat snps.txt 
ilmnseq_rs1799971_ilmndup1,ilmnseq_rs1799971_ilmndup2,rs1799971
ilmnseq_rs6265_ilmndup1,ilmnseq_rs6265_ilmndup2,rs6265
ilmnseq_rs1491850_ilmndup1,ilmnseq_rs1491850_ilmndup2,rs1491850</code></pre>
<p>The <code>plink</code> command can then use that file to create a table of the minor allele frequencies for a given set of genetic loci. –bfile is a very common flag in plink, as it is the input binary files (.bim, .bed, .fam) for a given filtering step.</p>
<pre class="shell"><code>$ ./plink \
  --bfile 2025-06-02_UCSD_GV_Genotypes_Runlists_1-5_QC_freeze2 \
  --extract snps.txt \
  --freq \
  --out snp_frequencies </code></pre>
<p>Here is the resulting table:</p>
<div class="cell">
<div class="cell-output-display">
<div class="kable-table">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 4%">
<col style="width: 71%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 7%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">CHR</th>
<th style="text-align: left;">SNP</th>
<th style="text-align: left;">A1</th>
<th style="text-align: left;">A2</th>
<th style="text-align: right;">MAF</th>
<th style="text-align: right;">NCHROBS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">6</td>
<td style="text-align: left;">ilmnseq_rs1799971_ilmndup1,ilmnseq_rs1799971_ilmndup2,rs1799971</td>
<td style="text-align: left;">G</td>
<td style="text-align: left;">A</td>
<td style="text-align: right;">0.1182</td>
<td style="text-align: right;">2750</td>
</tr>
<tr class="even">
<td style="text-align: right;">11</td>
<td style="text-align: left;">ilmnseq_rs6265_ilmndup1,ilmnseq_rs6265_ilmndup2,rs6265</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">0.1535</td>
<td style="text-align: right;">2750</td>
</tr>
<tr class="odd">
<td style="text-align: right;">11</td>
<td style="text-align: left;">ilmnseq_rs1491850_ilmndup1,ilmnseq_rs1491850_ilmndup2,rs1491850</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">T</td>
<td style="text-align: right;">0.3760</td>
<td style="text-align: right;">2750</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<ul>
<li>CHR = chromosome number</li>
<li>SNP = SNP identifier</li>
<li>A1 = minor (alternate) allele</li>
<li>A2 = major (reference) allele</li>
<li>MAF = minor allele frequency (calculated # minor allele observations / # total allele observations)</li>
<li>NCHROBS = total number of allele observations (2 * number of participants)</li>
</ul>
<hr>
</section>
</section>
<section id="quality-control" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="quality-control"><span class="header-section-number">1.3</span> Quality Control</h2>
<p>The below procedures were performed with these data:</p>
<p>RL1-5.bim</p>
<p>RL1-5.fam</p>
<p>RL1-5.bed</p>
<section id="overall-filters" class="level3" data-number="1.3.1">
<h3 data-number="1.3.1" class="anchored" data-anchor-id="overall-filters"><span class="header-section-number">1.3.1</span> Overall Filters</h3>
<ul>
<li><p>Participants with &gt; 2% missing genotype calls were removed. In cases of duplicated samples or sibling pairs, one individual from each pair was excluded to avoid overrepresentation of shared genotypes. The individual with lower data quality was preferentially dropped.</p></li>
<li><p>Variants with missing genotype calls in more than 20% of participants were excluded.</p></li>
<li><p>For the X and Y chromosomes, biologically impossible calls were removed — including heterozygous calls in males (who are hemizygous for these chromosomes) and non-missing Y chromosome calls in females.</p></li>
</ul>
</section>
<section id="considerations-while-working-with-these-data" class="level3" data-number="1.3.2">
<h3 data-number="1.3.2" class="anchored" data-anchor-id="considerations-while-working-with-these-data"><span class="header-section-number">1.3.2</span> Considerations while working with these data</h3>
<ul>
<li><p>We used a relatively liberal –geno 0.2 filter to retain as many variants as possible. Many studies adopt stricter thresholds (e.g., –geno 0.1 or lower), but we prioritized inclusivity of data in this stage of QC.</p></li>
<li><p>We intentionally did not apply a Hardy-Weinberg equilibrium (–hwe) filter. Principal component analysis (PCA) revealed population structure, and applying HWE filtering across pooled subpopulations can lead to inflated Type I error rates due to the Wahlund effect. This occurs when heterozygosity is overestimated across stratified populations (since there are actually more homozygotes in each subpopulation), causing variants to falsely appear as deviating from equilibrium. If desired, users may stratify the data by major ancestry groups and apply HWE filtering within those subsets.</p></li>
<li><p>We also chose not to filter on minor allele frequency (–maf) to leave this decision to the end user, depending on whether they are interested in rare or common variants. Common –maf thresholds include 0.05 or 0.01. Note that population structure can influence allele frequency distributions — a variant rare in the total sample may be common within a subpopulation.</p></li>
<li><p>The pseudo-autosomal region (PAR) of the X chromosome is often excluded in X chromosome-specific analyses, as it behaves more like an autosome. We leave this decision to researchers conducting such analyses. To perform this separation, a typical command is:</p></li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> path/to/x-chr <span class="at">--split-x</span> hg38 <span class="at">--make-bed</span> <span class="at">--out</span> path/to/split-off-PAR-x-chr</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>Many study designs, such as polygenic risk score calculations, must be cognizant of population structure.</li>
</ul>
</section>
<section id="quality-control-code" class="level3" data-number="1.3.3">
<h3 data-number="1.3.3" class="anchored" data-anchor-id="quality-control-code"><span class="header-section-number">1.3.3</span> Quality Control Code</h3>
<details>
<summary>
Click to show Quality Control Bash Script, which utilizes Plink 1.9
</summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> plots</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> helpers</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qc_derivatives</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Split files into individual chromosomes</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qc_derivatives/1_by_chr</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> chr <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">26</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> RL1-5 <span class="at">--chr</span> <span class="va">$chr</span> <span class="at">--make-bed</span> <span class="at">--out</span> qc_derivatives/1_by_chr/RL1-5_chr<span class="va">${chr}</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 2: Obtain Variant and Individual Level Summary Statistics on Each Chromosome's Data</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qc_derivatives/2_summary_stats</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> chr <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/1_by_chr/RL1-5_chr<span class="va">${chr}</span> <span class="at">--keep-allele-order</span> <span class="at">--missing</span> <span class="at">--freq</span> <span class="at">--hardy</span> <span class="at">--out</span> qc_derivatives/2_summary_stats/RL1-5_ss_chr<span class="va">${chr}</span><span class="kw">;</span> </span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="co"># 2.1: RScript to assess which participants &gt; 2% missingness </span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> ./scripts/missingness.samples.r <span class="st">"</span><span class="va">$USER_WD</span><span class="st">"</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 3: Drop individuals with high missingness</span></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qc_derivatives/3_ind_missingness_dropped</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> chr <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> </span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/1_by_chr/RL1-5_chr<span class="va">${chr}</span>  <span class="at">--remove</span> helpers/A2CPS_Ind_Missingness_Drop.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> qc_derivatives/3_ind_missingness_dropped/RL1-5_chr<span class="va">${chr}</span>_ind_missingness_dropped<span class="kw">;</span> </span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 4: Variant Quality Control based on Missingness (maf filtering avoided, users choice. hwe filtering avoided, wahlund effect)</span></span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qc_derivatives/4_variant_filtered</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> chr <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> </span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/3_ind_missingness_dropped/RL1-5_chr<span class="va">${chr}</span>_ind_missingness_dropped <span class="at">--keep-allele-order</span> <span class="at">--geno</span> 0.2 <span class="at">--write-snplist</span> <span class="at">--make-bed</span> <span class="at">--out</span> qc_derivatives/4_variant_filtered/RL1-5_chr<span class="va">${chr}</span>_variant_filtered<span class="kw">;</span> </span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 5 / 6: LD Pruning</span></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qc_derivatives/5_LD_Pruning</span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> chr <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> </span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/4_variant_filtered/RL1-5_chr<span class="va">${chr}</span>_variant_filtered <span class="at">--indep-pairwise</span> 100 5 0.2 <span class="at">--out</span> qc_derivatives/5_LD_Pruning/RL1-5_chr<span class="va">${chr}</span>_LD_Pruning<span class="kw">;</span> </span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qc_derivatives/6_LD_Pruned</span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> chr <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> </span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/4_variant_filtered/RL1-5_chr<span class="va">${chr}</span>_variant_filtered <span class="at">--extract</span> qc_derivatives/5_LD_Pruning/RL1-5_chr<span class="va">${chr}</span>_LD_Pruning.prune.in <span class="at">--make-bed</span> <span class="at">--out</span> qc_derivatives/6_LD_Pruned/RL1-5_chr<span class="va">${chr}</span>_LD_Pruned<span class="kw">;</span> </span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 7: Merge</span></span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qc_derivatives/7_Merged</span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> <span class="bu">echo</span> qc_derivatives/6_LD_Pruned/RL1-5_chr<span class="va">${i}</span>_LD_Pruned<span class="kw">;</span> <span class="cf">done</span> <span class="op">&gt;</span> helpers/RL1-5_chr_LD_Pruned_list.txt</span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/6_LD_Pruned/RL1-5_chr1_LD_Pruned <span class="at">--merge-list</span> helpers/RL1-5_chr_LD_Pruned_list.txt <span class="at">--make-bed</span> <span class="at">--out</span> qc_derivatives/7_Merged/RL1-5_Merged</span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 8: Heterozygosity Checks</span></span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/7_Merged/RL1-5_Merged <span class="at">--het</span> <span class="at">--out</span> qc_derivatives/7_Merged/RL1-5_Merged_Het</span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 9: Identity by Descent Checks - if related or duplicates exist, one of the family-members (the one with worse quality) must be dropped.</span></span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/7_Merged/RL1-5_Merged <span class="at">--genome</span> <span class="at">--threads</span> 1 <span class="at">--out</span> qc_derivatives/7_Merged/RL1-5_Merged_IBD</span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a><span class="co"># 9.1: RScripts to identify participants to drop based on heterozygosity and poorer-quality related sample</span></span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> ./scripts/het.genome.r <span class="st">"</span><span class="va">$USER_WD</span><span class="st">"</span></span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> ./scripts/missingness.related.comparison.r <span class="st">"</span><span class="va">$USER_WD</span><span class="st">"</span></span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a><span class="co">### Step 10: Sex Typing </span><span class="al">###</span></span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> qc_derivatives/sex_typing</span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create list of individuals to be dropped prior to sex typing, those with high missingness and poorer-quality related individuals</span></span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span> helpers/A2CPS_Ind_Missingness_Drop.txt helpers/A2CPS_related_drop.txt  <span class="op">&gt;</span> helpers/A2CPS_Pre_Sex_Typing_Sample_Drops.txt</span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step a: Drop Samples that did not pass QC</span></span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> qc_derivatives/sex_typing/samples_dropped</span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/1_by_chr/RL1-5_chr23 <span class="at">--remove</span> helpers/A2CPS_Pre_Sex_Typing_Sample_Drops.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> qc_derivatives/sex_typing/samples_dropped/RL1-5_chr23_samples_dropped<span class="kw">;</span> </span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step b: Split off X Chr pseudo-autosomal region (whose inheritance patterns mirror those of the autosomes) [chr25]</span></span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> qc_derivatives/sex_typing/PAR_Removed</span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/sex_typing/samples_dropped/RL1-5_chr23_samples_dropped <span class="at">--split-x</span> hg38 <span class="at">--make-bed</span> <span class="at">--out</span> qc_derivatives/sex_typing/PAR_Removed/RL1-5_chr23_PAR_Removed</span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step c: LD Pruning </span></span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> qc_derivatives/sex_typing/LD_Pruning</span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/sex_typing/PAR_Removed/RL1-5_chr23_PAR_Removed <span class="at">--indep-pairphase</span> 100 5 0.2 <span class="at">--out</span> qc_derivatives/sex_typing/LD_Pruning/RL1-5_chr23_LD_Pruning</span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> qc_derivatives/sex_typing/LD_Pruned</span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/sex_typing/PAR_Removed/RL1-5_chr23_PAR_Removed <span class="at">--extract</span> qc_derivatives/sex_typing/LD_Pruning/RL1-5_chr23_LD_Pruning.prune.in <span class="at">--make-bed</span> <span class="at">--out</span> qc_derivatives/sex_typing/LD_Pruned/RL1-5_chr23_LD_Pruned</span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step d: MAF Estimates</span></span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> qc_derivatives/sex_typing/MAF</span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/sex_typing/LD_Pruned/RL1-5_chr23_LD_Pruned <span class="at">--freq</span> <span class="at">--out</span> qc_derivatives/sex_typing/MAF/RL1-5_chr23_LD_Pruned_MAF</span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step e: Sex Check</span></span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/sex_typing/LD_Pruned/RL1-5_chr23_LD_Pruned  <span class="at">--read-freq</span> qc_derivatives/sex_typing/MAF/RL1-5_chr23_LD_Pruned_MAF.frq <span class="at">--check-sex</span> 0.3 0.75 <span class="at">--out</span> qc_derivatives/sex_typing/sex_check</span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step f: Create Plots and create list of participants to drop, where we were unable to resolve the sex discrepancies with REDCap or EHR data.</span></span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a>    <span class="ex">Rscript</span> ./scripts/sex.typing.r <span class="st">"</span><span class="va">$USER_WD</span><span class="st">"</span></span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 11: Return to pre-LD-Pruned files and merge. </span></span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> qc_derivatives/8_Merged_Pre_LD</span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> <span class="bu">echo</span> qc_derivatives/4_variant_filtered/RL1-5_chr<span class="va">${i}</span>_variant_filtered<span class="kw">;</span> <span class="cf">done</span> <span class="op">&gt;</span> helpers/RL1-5_chr_Pre_LD_list.txt</span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/4_variant_filtered/RL1-5_chr1_variant_filtered <span class="at">--merge-list</span> helpers/RL1-5_chr_Pre_LD_list.txt <span class="at">--make-bed</span> <span class="at">--out</span> qc_derivatives/8_Merged_Pre_LD/RL1-5_Merged_Pre_LD</span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a><span class="co"># 11.1: Create comprehensive file with all participant drops so far</span></span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> helpers/A2CPS_Ind_Missingness_Drop.txt helpers/A2CPS_related_and_sex_discrepancy_drop.txt  <span class="op">&gt;</span> helpers/A2CPS_Pre_Chr_XYMT_QC_Sample_Drops.txt</span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a><span class="co">### X, Y, and MT QC </span><span class="al">###</span></span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a><span class="co">### X Chromosome QC </span><span class="al">###</span></span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> X_Chr</span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step a: drop samples with high missingness, higher-missingness related samples, and unrecoverable sex typing discrepancies, fix the sex discrepancies</span></span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> X_Chr/Samples_Dropped</span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/1_by_chr/RL1-5_chr23 <span class="at">--remove</span> helpers/A2CPS_Pre_Chr_XYMT_QC_Sample_Drops.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> X_Chr/Samples_Dropped/RL1-5_chr23_samples_dropped</span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a>    <span class="co"># step b: update sex</span></span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> X_Chr/Sex_Updated</span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span>  X_Chr/Samples_Dropped/RL1-5_chr23_samples_dropped <span class="at">--update-sex</span> helpers/A2CPS_update_sex.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> X_Chr/Sex_Updated/RL1-5_chr23_Sex_Updated</span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step c: Take summary statistics in case necessary to further remove samples</span></span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> X_Chr/Summary_Stats</span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> X_Chr/Sex_Updated/RL1-5_chr23_Sex_Updated <span class="at">--missing</span> <span class="at">--freq</span> <span class="at">--out</span> X_Chr/Summary_Stats/RL1-5_chr23_Summary_Stats</span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step d: Filter out poor-quality variants</span></span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> X_Chr/Variant_Filtered</span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> X_Chr/Sex_Updated/RL1-5_chr23_Sex_Updated <span class="at">--geno</span> 0.2 <span class="at">--write-snplist</span> <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> X_Chr/Variant_Filtered/RL1-5_chr23_variant_filtered</span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a>    <span class="fu">awk</span> <span class="st">'{print $3}'</span> X_Chr/Variant_Filtered/RL1-5_chr23_variant_filtered.hh <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-u</span> <span class="op">&gt;</span> X_Chr/Variant_Filtered/hh_snps_x_to_exclude.txt</span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a>    <span class="co"># step e: Filter out those included in the .hh file</span></span>
<span id="cb8-119"><a href="#cb8-119" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grep</span> <span class="at">-vFf</span> X_Chr/Variant_Filtered/hh_snps_x_to_exclude.txt X_Chr/Variant_Filtered/RL1-5_chr23_variant_filtered.snplist <span class="op">&gt;</span> X_Chr/Variant_Filtered/RL1-5_chr23_variant_filtered.no_hh.snplist</span>
<span id="cb8-120"><a href="#cb8-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-121"><a href="#cb8-121" aria-hidden="true" tabindex="-1"></a><span class="co">### Y Chromosome QC </span><span class="al">###</span></span>
<span id="cb8-122"><a href="#cb8-122" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> Y_Chr</span>
<span id="cb8-123"><a href="#cb8-123" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step a: drop samples with high missingness, higher-missingness related samples, and unrecoverable sex typing discrepancies </span></span>
<span id="cb8-124"><a href="#cb8-124" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> Y_Chr/Samples_Dropped</span>
<span id="cb8-125"><a href="#cb8-125" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/1_by_chr/RL1-5_chr24 <span class="at">--remove</span> helpers/A2CPS_Pre_Chr_XYMT_QC_Sample_Drops.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> Y_Chr/Samples_Dropped/RL1-5_chr24_samples_dropped</span>
<span id="cb8-126"><a href="#cb8-126" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-127"><a href="#cb8-127" aria-hidden="true" tabindex="-1"></a>    <span class="co"># step b: update sex</span></span>
<span id="cb8-128"><a href="#cb8-128" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> Y_Chr/Sex_Updated</span>
<span id="cb8-129"><a href="#cb8-129" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span>  Y_Chr/Samples_Dropped/RL1-5_chr24_samples_dropped <span class="at">--update-sex</span> helpers/A2CPS_update_sex.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> Y_Chr/Sex_Updated/RL1-5_chr24_Sex_Updated</span>
<span id="cb8-130"><a href="#cb8-130" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-131"><a href="#cb8-131" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step c: Take summary statistics in case necessary to further remove samples</span></span>
<span id="cb8-132"><a href="#cb8-132" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> Y_Chr/Summary_Stats</span>
<span id="cb8-133"><a href="#cb8-133" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> Y_Chr/Sex_Updated/RL1-5_chr24_Sex_Updated <span class="at">--missing</span> <span class="at">--freq</span> <span class="at">--out</span> Y_Chr/Summary_Stats/RL1-5_chr24_Summary_Stats</span>
<span id="cb8-134"><a href="#cb8-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-135"><a href="#cb8-135" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step d: Filter out poor-quality variants</span></span>
<span id="cb8-136"><a href="#cb8-136" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> Y_Chr/Variant_Filtered</span>
<span id="cb8-137"><a href="#cb8-137" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> Y_Chr/Sex_Updated/RL1-5_chr24_Sex_Updated <span class="at">--geno</span> 0.2 <span class="at">--keep-allele-order</span> <span class="at">--write-snplist</span> <span class="at">--make-bed</span> <span class="at">--out</span> Y_Chr/Variant_Filtered/RL1-5_chr24_variant_filtered</span>
<span id="cb8-138"><a href="#cb8-138" aria-hidden="true" tabindex="-1"></a>    <span class="fu">awk</span> <span class="st">'{print $3}'</span> Y_Chr/Variant_Filtered/RL1-5_chr24_variant_filtered.hh <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-u</span> <span class="op">&gt;</span> Y_Chr/Variant_Filtered/hh_snps_y_to_exclude.txt</span>
<span id="cb8-139"><a href="#cb8-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-140"><a href="#cb8-140" aria-hidden="true" tabindex="-1"></a>    <span class="co"># step e: Filter out those included in the .hh file</span></span>
<span id="cb8-141"><a href="#cb8-141" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grep</span> <span class="at">-vFf</span> Y_Chr/Variant_Filtered/hh_snps_y_to_exclude.txt Y_Chr/Variant_Filtered/RL1-5_chr24_variant_filtered.snplist <span class="op">&gt;</span> Y_Chr/Variant_Filtered/RL1-5_chr24_variant_filtered.no_hh.snplist</span>
<span id="cb8-142"><a href="#cb8-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-143"><a href="#cb8-143" aria-hidden="true" tabindex="-1"></a><span class="co">### Mitochondrial DNA </span><span class="al">###</span></span>
<span id="cb8-144"><a href="#cb8-144" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> MT_Chr</span>
<span id="cb8-145"><a href="#cb8-145" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step a: drop samples with high missingness, higher-missingness related samples, and unrecoverable sex typing discrepancies </span></span>
<span id="cb8-146"><a href="#cb8-146" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> MT_Chr/Samples_Dropped</span>
<span id="cb8-147"><a href="#cb8-147" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> qc_derivatives/1_by_chr/RL1-5_chr26 <span class="at">--remove</span> helpers/A2CPS_Pre_Chr_XYMT_QC_Sample_Drops.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> MT_Chr/Samples_Dropped/RL1-5_chr26_samples_dropped</span>
<span id="cb8-148"><a href="#cb8-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-149"><a href="#cb8-149" aria-hidden="true" tabindex="-1"></a>    <span class="co"># step b: update sex</span></span>
<span id="cb8-150"><a href="#cb8-150" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> MT_Chr/Sex_Updated</span>
<span id="cb8-151"><a href="#cb8-151" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span>  MT_Chr/Samples_Dropped/RL1-5_chr26_samples_dropped <span class="at">--update-sex</span> helpers/A2CPS_update_sex.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> MT_Chr/Sex_Updated/RL1-5_chr26_Sex_Updated</span>
<span id="cb8-152"><a href="#cb8-152" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-153"><a href="#cb8-153" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step c: Take summary statistics in case necessary to further remove samples</span></span>
<span id="cb8-154"><a href="#cb8-154" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> MT_Chr/Summary_Stats</span>
<span id="cb8-155"><a href="#cb8-155" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> MT_Chr/Sex_Updated/RL1-5_chr26_Sex_Updated <span class="at">--missing</span> <span class="at">--freq</span> <span class="at">--out</span> MT_Chr/Summary_Stats/RL1-5_chr26_Summary_Stats</span>
<span id="cb8-156"><a href="#cb8-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-157"><a href="#cb8-157" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step d: Filter out poor-quality variants</span></span>
<span id="cb8-158"><a href="#cb8-158" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> MT_Chr/Variant_Filtered</span>
<span id="cb8-159"><a href="#cb8-159" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> MT_Chr/Sex_Updated/RL1-5_chr26_Sex_Updated <span class="at">--geno</span> 0.2 <span class="at">--keep-allele-order</span> <span class="at">--write-snplist</span> <span class="at">--make-bed</span> <span class="at">--out</span> MT_Chr/Variant_Filtered/RL1-5_chr26_variant_filtered</span>
<span id="cb8-160"><a href="#cb8-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-161"><a href="#cb8-161" aria-hidden="true" tabindex="-1"></a><span class="co">## LIST OF INDIVIDUALS TO DROP ##</span></span>
<span id="cb8-162"><a href="#cb8-162" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> helpers/A2CPS_Ind_Missingness_Drop.txt helpers/A2CPS_related_and_sex_discrepancy_drop.txt <span class="op">&gt;</span> A2CPS_RL1-5_Preparation_Files/A2CPS_RL1-5_Final_Sample_Drop_List.txt</span>
<span id="cb8-163"><a href="#cb8-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-164"><a href="#cb8-164" aria-hidden="true" tabindex="-1"></a><span class="co">## LIST OF SNPs TO KEEP ##</span></span>
<span id="cb8-165"><a href="#cb8-165" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> qc_derivatives/4_variant_filtered/RL1-5_chr{1..22}_variant_filtered.snplist X_Chr/Variant_Filtered/RL1-5_chr23_variant_filtered.no_hh.snplist Y_Chr/Variant_Filtered/RL1-5_chr24_variant_filtered.no_hh.snplist  MT_Chr/Variant_Filtered/RL1-5_chr26_variant_filtered.snplist <span class="op">&gt;</span> A2CPS_RL1-5_Preparation_Files/RL1-5_all_chr_variant_filtered.snplist</span>
<span id="cb8-166"><a href="#cb8-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-167"><a href="#cb8-167" aria-hidden="true" tabindex="-1"></a><span class="co">## LIST OF SEXES TO UPDATE ##</span></span>
<span id="cb8-168"><a href="#cb8-168" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> helpers/A2CPS_update_sex.txt A2CPS_RL1-5_Preparation_Files/A2CPS_RL1-5_Update_Sex_List.txt </span>
<span id="cb8-169"><a href="#cb8-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-170"><a href="#cb8-170" aria-hidden="true" tabindex="-1"></a><span class="co"># USE THIS LINE TO OBTAIN FINAL FILES QC FIles, still with participants outside of Freeze 2 ##</span></span>
<span id="cb8-171"><a href="#cb8-171" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> ./A2CPS_RL1-5_QCed</span>
<span id="cb8-172"><a href="#cb8-172" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> RL1-5 <span class="at">--remove</span> A2CPS_RL1-5_Preparation_Files/A2CPS_RL1-5_Final_Sample_Drop_List.txt <span class="at">--extract</span> A2CPS_RL1-5_Preparation_Files/RL1-5_all_chr_variant_filtered.snplist <span class="at">--update-sex</span> A2CPS_RL1-5_Preparation_Files/A2CPS_RL1-5_Update_Sex_List.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> A2CPS_RL1-5_QCed/A2CPS_RL1-5_QCed</span>
<span id="cb8-173"><a href="#cb8-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-174"><a href="#cb8-174" aria-hidden="true" tabindex="-1"></a><span class="co">########################</span></span>
<span id="cb8-175"><a href="#cb8-175" aria-hidden="true" tabindex="-1"></a><span class="co">## FINAL FOR RELEASE: ##</span></span>
<span id="cb8-176"><a href="#cb8-176" aria-hidden="true" tabindex="-1"></a><span class="co">########################</span></span>
<span id="cb8-177"><a href="#cb8-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-178"><a href="#cb8-178" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset only Freeze 2 #</span></span>
<span id="cb8-179"><a href="#cb8-179" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> ./A2CPS_RL1-5_QCed/Freeze2.R <span class="st">"</span><span class="va">$USER_WD</span><span class="st">"</span></span>
<span id="cb8-180"><a href="#cb8-180" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> Freeze_2</span>
<span id="cb8-181"><a href="#cb8-181" aria-hidden="true" tabindex="-1"></a><span class="ex">./plink</span> <span class="at">--bfile</span> RL1-5 <span class="at">--remove</span> A2CPS_RL1-5_Preparation_Files/A2CPS_RL1-5_Final_Sample_Drop_List_Only_Freeze_2.txt <span class="at">--extract</span> A2CPS_RL1-5_Preparation_Files/RL1-5_all_chr_variant_filtered.snplist <span class="at">--update-sex</span> A2CPS_RL1-5_Preparation_Files/A2CPS_RL1-5_Update_Sex_List.txt <span class="at">--keep-allele-order</span> <span class="at">--make-bed</span> <span class="at">--out</span> Freeze_2/A2CPS_Freeze_2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<section id="supplementary-qc-files-rscripts" class="level4" data-number="1.3.3.1">
<h4 data-number="1.3.3.1" class="anchored" data-anchor-id="supplementary-qc-files-rscripts"><span class="header-section-number">1.3.3.1</span> Supplementary QC Files: RScripts</h4>
<details>
<summary>
Click to show Supplementary RScript #1: Determining High Missingness Individuals (missingness.samples.r)
</summary>
<div class="sourceCode" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>combined_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>user_wd <span class="ot">&lt;-</span> args[<span class="dv">1</span>]</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(user_wd)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co"># read in and combine .imiss file (individual-level missingness)</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>) {</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  file_name <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">"./qc_derivatives/2_summary_stats/RL1-5_ss_chr%d.imiss"</span>, i)</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  current_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(file_name, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  combined_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(combined_data, current_data)</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co"># DAT HAS EACH MISSING CALL RATE (22 per participant)</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> combined_data<span class="sc">$</span><span class="st">"F_MISS"</span> </span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>number.geno <span class="ot">&lt;-</span> combined_data<span class="sc">$</span>N_GENO <span class="co"># total genotyped SNPs across autosomes</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>number.missed <span class="ot">&lt;-</span> combined_data<span class="sc">$</span>N_MISS <span class="co"># total missed SNPs across autosomes across participants</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>matrix.geno <span class="ot">&lt;-</span> <span class="fu">matrix</span>(number.geno,<span class="at">ncol=</span><span class="dv">22</span>)</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>summed.geno <span class="ot">&lt;-</span> <span class="fu">sum</span>(matrix.geno[<span class="dv">1</span>,]) <span class="co"># total autosomal SNPs</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>matrix.missed <span class="ot">&lt;-</span> <span class="fu">matrix</span>(number.missed,<span class="at">ncol=</span><span class="dv">22</span>)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>individual.missed.percent <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(matrix.missed) <span class="sc">/</span> summed.geno</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>missingness.table <span class="ot">&lt;-</span>  <span class="fu">data.table</span>(</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">FAMID =</span> current_data<span class="sc">$</span>FID,</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">ID =</span> current_data<span class="sc">$</span>IID,</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">Missingness =</span> individual.missed.percent</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(missingness.table, <span class="st">"./helpers/A2CPS_RL1-5_Missingness.csv"</span>)</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a><span class="co"># subset those with over %2 missingness</span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>over.two <span class="ot">&lt;-</span> <span class="fu">which</span>(individual.missed.percent <span class="sc">&gt;</span> <span class="fl">0.02</span>)</span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>over.two.values <span class="ot">&lt;-</span> individual.missed.percent[over.two]</span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>over.two.missing.SNPs <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(matrix.missed)[over.two]</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>subject_ID <span class="ot">&lt;-</span> combined_data<span class="sc">$</span>IID[over.two]</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>high.ind.missingness <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>  <span class="at">ID =</span> over.two,</span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>  <span class="at">SUBJECT_ID =</span> subject_ID,</span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>  <span class="at">MISSINGNESS =</span> over.two.values,</span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>  <span class="at">SNPs_MISSED =</span> over.two.missing.SNPs</span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a><span class="co"># table with high missingness individuals for PLINK Dropping</span></span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>for_drop <span class="ot">&lt;-</span> high.ind.missingness[, .(ID, SUBJECT_ID)]</span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(for_drop, <span class="st">"./helpers/A2CPS_Ind_Missingness_Drop.txt"</span>, <span class="at">sep =</span> <span class="st">" "</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a><span class="co"># spaghetti plots across chromosomes</span></span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>r <span class="ot">&lt;-</span> <span class="fu">range</span>(dat)</span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"./plots/missingness.samples.perchr.spaghetti.A2CPS.pdf"</span>,<span class="at">width=</span><span class="dv">9</span>,<span class="at">height=</span><span class="dv">7</span>)</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">las=</span><span class="dv">1</span>)</span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">matrix</span>(dat,<span class="at">ncol=</span><span class="dv">22</span>)</span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">22</span>),r,<span class="at">type=</span><span class="st">"n"</span>,<span class="at">xlab=</span><span class="st">"chromosome"</span>,<span class="at">ylab=</span><span class="st">"% missing"</span>,<span class="at">xaxt=</span><span class="st">"n"</span>)</span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">axis</span>(<span class="dv">1</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>)</span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(tmp)) <span class="fu">lines</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>,tmp[j,],<span class="at">col=</span><span class="st">"orange"</span>)</span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a><span class="co"># boxplot sample average</span></span>
<span id="cb9-61"><a href="#cb9-61" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">matrix</span>(dat,<span class="at">ncol=</span><span class="dv">22</span>) <span class="co">#temp is a matrix with rows as participants and columns as chr## and cells as FMISS </span></span>
<span id="cb9-62"><a href="#cb9-62" aria-hidden="true" tabindex="-1"></a>z.pm <span class="ot">&lt;-</span> <span class="fu">apply</span>(tmp,<span class="dv">1</span>,mean) <span class="co"># z.pm is a vector with the mean values of FMISS for each participant ie. what is their average FMISS rate across participants</span></span>
<span id="cb9-63"><a href="#cb9-63" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"./plots/missingness.samples.averages.A2CPS.pdf"</span>,<span class="at">width=</span><span class="dv">9</span>,<span class="at">height=</span><span class="dv">6</span>)</span>
<span id="cb9-64"><a href="#cb9-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="at">las=</span><span class="dv">1</span>)</span>
<span id="cb9-65"><a href="#cb9-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">boxplot</span>(z.pm)</span>
<span id="cb9-66"><a href="#cb9-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">boxplot</span>(z.pm,<span class="at">log=</span><span class="st">"y"</span>)</span>
<span id="cb9-67"><a href="#cb9-67" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb9-68"><a href="#cb9-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-69"><a href="#cb9-69" aria-hidden="true" tabindex="-1"></a><span class="co"># percentile missingness plot</span></span>
<span id="cb9-70"><a href="#cb9-70" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">length</span>(z.pm)</span>
<span id="cb9-71"><a href="#cb9-71" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"./plots/missingness.samples.percentile.A2CPS.pdf"</span>,<span class="at">width=</span><span class="dv">8</span>,<span class="at">height=</span><span class="dv">7</span>)</span>
<span id="cb9-72"><a href="#cb9-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>),<span class="at">las=</span><span class="dv">1</span>)</span>
<span id="cb9-73"><a href="#cb9-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">c</span>(<span class="fl">0.9</span>,<span class="dv">1</span>),<span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.4</span>),<span class="at">type=</span><span class="st">"n"</span>,<span class="at">xlab=</span><span class="st">"percentile missingness"</span>,<span class="at">ylab=</span><span class="st">"% missing"</span>)</span>
<span id="cb9-74"><a href="#cb9-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="fl">0.1</span>,<span class="fl">0.01</span>),<span class="at">lty=</span><span class="st">"dotted"</span>,<span class="at">col=</span><span class="st">"lightgrey"</span>)</span>
<span id="cb9-75"><a href="#cb9-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lines</span>((<span class="dv">1</span><span class="sc">:</span>n)<span class="sc">/</span>n,<span class="fu">sort</span>(z.pm),<span class="at">type=</span><span class="st">"b"</span>,<span class="at">cex=</span><span class="fl">0.5</span>,<span class="at">pch=</span><span class="dv">20</span>)</span>
<span id="cb9-76"><a href="#cb9-76" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<details>
<summary>
Click to show Supplementary RScript #2: Heterozygosity and Relatedness Checks (het.genome.r)
</summary>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>user_wd <span class="ot">&lt;-</span> args[<span class="dv">1</span>]</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(user_wd)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co"># ---------- </span><span class="al">TESTING</span><span class="co"> HETEROZYGOSITY ------------</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>a2cps.h <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./qc_derivatives/7_Merged/RL1-5_Merged_Het.het"</span>, <span class="at">stringsAsFactors =</span> F)</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(a2cps.h) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"FID"</span>,<span class="st">"IID"</span>,<span class="st">"O_HOM"</span>, <span class="st">"E_HOM"</span>, <span class="st">"N_NM"</span>,  <span class="st">"F"</span>)</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>a2cps.h[,<span class="st">'Het_Rate'</span> <span class="sc">:</span><span class="er">=</span> (N_NM <span class="sc">-</span> O_HOM)<span class="sc">/</span>N_NM]</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>a2cps.h[, <span class="st">'Pop'</span> <span class="sc">:</span><span class="er">=</span> <span class="st">"A2CPS"</span>]</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(a2cps.h)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>p.ht <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>dat2, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(Pop, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"A2CPS"</span>)), <span class="at">y=</span>Het_Rate, <span class="at">group=</span>Pop, <span class="at">color=</span>Pop)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="at">outlier.colour =</span> <span class="st">"red"</span>, <span class="at">show.legend =</span> F) <span class="sc">+</span> <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Populations"</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Het Rate"</span>, <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">0.0</span>,<span class="fl">0.32</span>)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Heterozygosity Rates"</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>, <span class="at">color=</span><span class="st">"black"</span>,<span class="at">hjust=</span><span class="dv">1</span>), <span class="at">axis.text.y=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>, <span class="at">color =</span> <span class="st">"black"</span>), <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">18</span>, <span class="at">face =</span> <span class="st">"bold"</span>), <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">18</span>, <span class="at">face =</span> <span class="st">"bold"</span>), <span class="at">panel.background=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.minor=</span><span class="fu">element_blank</span>(), <span class="at">panel.grid.major=</span><span class="fu">element_blank</span>(), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size=</span><span class="dv">20</span>, <span class="at">face =</span> <span class="st">"bold"</span>))</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"./plots/A2CPS_RL1-5_Het.png"</span>, p.ht, <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="co"># ---------- </span><span class="al">TESTING</span><span class="co"> PARENTAL RELATEDNESS ------------</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>philippines <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./qc_derivatives/7_Merged/RL1-5_Merged_IBD.genome"</span>, <span class="at">stringsAsFactors =</span> F)</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>philippines <span class="ot">&lt;-</span> philippines[<span class="fu">which</span>(philippines<span class="sc">$</span>Z0 <span class="sc">!=</span> <span class="dv">1</span>)]</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"./plots/A2CPS_RL1-5_Relatedness.png"</span>, <span class="at">width=</span><span class="dv">3000</span>, <span class="at">height=</span><span class="dv">2000</span>, <span class="at">res=</span><span class="dv">300</span>)</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">pch=</span><span class="dv">16</span>)</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">with</span>(philippines, <span class="fu">plot</span>(Z0, Z1, <span class="at">xlim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">type=</span><span class="st">"n"</span>, <span class="at">main=</span><span class="st">"A2CPS Z1 &gt; 0 Relatedness"</span>))</span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">with</span>(<span class="fu">subset</span>(philippines, RT <span class="sc">==</span> <span class="st">"UN"</span>), <span class="fu">points</span>(Z0, Z1, <span class="at">col=</span><span class="dv">1</span>, <span class="at">cex=</span><span class="fl">0.4</span>))</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>related <span class="ot">&lt;-</span> philippines[<span class="fu">which</span>(philippines<span class="sc">$</span>Z0 <span class="sc">&lt;</span> <span class="fl">0.5</span>), ]</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(related, <span class="st">"./helpers/A2CPS_RL1-5_related.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<details>
<summary>
Click to show Supplementary RScript #3: Determining which Related Participants to Drop (missingness.related.comparison.r)
</summary>
<div class="sourceCode" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>user_wd <span class="ot">&lt;-</span> args[<span class="dv">1</span>]</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(user_wd)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>missingness.table <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./helpers/A2CPS_RL1-5_Missingness.csv"</span>)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>related <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./helpers/A2CPS_RL1-5_related.csv"</span>)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>related_IDs_row1 <span class="ot">&lt;-</span> related<span class="sc">$</span>FID1</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>related_IDs_row2 <span class="ot">&lt;-</span> related<span class="sc">$</span>FID2</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>related_missingness_row1 <span class="ot">&lt;-</span> missingness.table[<span class="fu">which</span>(missingness.table<span class="sc">$</span>FAMID <span class="sc">%in%</span> related_IDs_row1), ]</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>related_missingness_row2 <span class="ot">&lt;-</span> missingness.table[<span class="fu">which</span>(missingness.table<span class="sc">$</span>FAMID <span class="sc">%in%</span> related_IDs_row2), ]</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder to match related_IDs_row1 and related_IDs_row2</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>comp1 <span class="ot">&lt;-</span> related_missingness_row1[<span class="fu">match</span>(related_IDs_row1, related_missingness_row1<span class="sc">$</span>FAMID)]</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>comp2 <span class="ot">&lt;-</span> related_missingness_row2[<span class="fu">match</span>(related_IDs_row2, related_missingness_row2<span class="sc">$</span>FAMID)]</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>complist <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ind <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(comp1))) {</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  related_pair <span class="ot">&lt;-</span> <span class="fu">c</span>(comp1<span class="sc">$</span>FAMID[ind], comp2<span class="sc">$</span>FAMID[ind])</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>  complist[[ind]] <span class="ot">&lt;-</span> related_pair</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>drop.list <span class="ot">&lt;-</span> <span class="fu">vector</span>()</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pair <span class="cf">in</span> complist) {</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>  missingness<span class="fl">.1</span> <span class="ot">&lt;-</span> missingness.table<span class="sc">$</span>Missingness[pair[<span class="dv">1</span>]]</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>  missingness<span class="fl">.2</span> <span class="ot">&lt;-</span> missingness.table<span class="sc">$</span>Missingness[pair[<span class="dv">2</span>]]</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (missingness<span class="fl">.1</span> <span class="sc">&gt;</span> missingness<span class="fl">.2</span>) {</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>    drop.list <span class="ot">&lt;-</span> <span class="fu">c</span>(drop.list, pair[<span class="dv">1</span>])</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>    drop.list <span class="ot">&lt;-</span> <span class="fu">c</span>(drop.list, pair[<span class="dv">2</span>])</span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>IID <span class="ot">&lt;-</span> missingness.table[<span class="fu">match</span>(drop.list, missingness.table<span class="sc">$</span>FAMID), ]<span class="sc">$</span>ID</span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>drop.table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>  FAMID <span class="ot">&lt;-</span> drop.list,</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>  IID <span class="ot">&lt;-</span> IID</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Drop the higher-missingness related / duplicate individual</span></span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(drop.table, <span class="at">sep =</span> <span class="st">" "</span>, <span class="at">col.names =</span>  F,<span class="st">"./helpers/A2CPS_related_drop.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<details>
<summary>
Click to show Supplementary RScript #4: Sex Typing Discrepancy Analysis and Handling of Discrepancies (sex.typing.r)
</summary>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>user_wd <span class="ot">&lt;-</span> args[<span class="dv">1</span>]</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(user_wd)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>st_Info <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"qc_derivatives/sex_typing/sex_check.sexcheck"</span>)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>st_Info_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(st_Info)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>ok_df <span class="ot">&lt;-</span> st_Info_df[st_Info_df<span class="sc">$</span>STATUS <span class="sc">==</span> <span class="st">"OK"</span>, ]</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>problem_df <span class="ot">&lt;-</span> st_Info_df[st_Info_df<span class="sc">$</span>STATUS <span class="sc">==</span> <span class="st">"PROBLEM"</span>, ]</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>info1_4 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./demographic_data/2025-01-13_UCSD_GV_Genotypes_Runlists_1-4_QC.csv"</span>)[, <span class="fu">c</span>(<span class="st">"subject_guid"</span>,<span class="st">"individual_file_tag"</span>)]</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>info5 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./demographic_data/2025-02-11_UCSD_GV_Genotypes_Runlist_5_QC.csv"</span>)[, <span class="fu">c</span>(<span class="st">"subject_guid"</span>,<span class="st">"individual_file_tag"</span>)]</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>info_all <span class="ot">&lt;-</span> <span class="fu">rbind</span>(info1_4, info5)</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>problem_IIDs <span class="ot">&lt;-</span> problem_df<span class="sc">$</span>IID <span class="do">## The subject IDs of the problematic samples</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>info_problem_IIDs <span class="ot">&lt;-</span> info_all[<span class="fu">which</span>(info_all<span class="sc">$</span>individual_file_tag <span class="sc">%in%</span> problem_IIDs), ]</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>merged_dt <span class="ot">&lt;-</span> <span class="fu">merge</span>(</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>  problem_df, </span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>  info_problem_IIDs, </span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">by.x =</span> <span class="st">"IID"</span>, </span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">by.y =</span> <span class="st">"individual_file_tag"</span>, </span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">all.x =</span> <span class="cn">TRUE</span>  </span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a><span class="fu">setDT</span>(merged_dt)</span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>merged_dt[, subject_guid <span class="sc">:</span><span class="er">=</span> <span class="fu">toupper</span>(<span class="fu">as.character</span>(subject_guid))]</span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>redcap <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./demographic_data/REDCap/demographics-2025-01-10 (1).csv"</span>)</span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>redcap.genident <span class="ot">&lt;-</span> redcap<span class="sc">$</span>genident</span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>redcap.sex <span class="ot">&lt;-</span> redcap<span class="sc">$</span>sex</span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a><span class="fu">setDT</span>(redcap)</span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a>merged_dt <span class="ot">&lt;-</span> redcap[, .(guid, genident)][merged_dt, on <span class="ot">=</span> .(<span class="at">guid =</span> subject_guid)]</span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a>merged_dt <span class="ot">&lt;-</span> redcap[, .(guid, sex)][merged_dt, on <span class="ot">=</span> .(<span class="at">guid =</span> guid)]</span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a>ehr1 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./demographic_data/EHR/reformatted_ns_clinicals_and_case_characteristics.csv"</span>)[, <span class="fu">c</span>(<span class="st">"guid"</span>, <span class="st">"sex_ehr"</span>)]</span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>ehr2 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./demographic_data/EHR/reformatted_ru_clinicals_and_case_characteristics.csv"</span>)[, <span class="fu">c</span>(<span class="st">"guid"</span>, <span class="st">"sex_ehr"</span>)]</span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a>ehr3 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./demographic_data/EHR/reformatted_um_clinicals_and_case_characteristics.csv"</span>)[, <span class="fu">c</span>(<span class="st">"guid"</span>, <span class="st">"sex_ehr"</span>)]</span>
<span id="cb12-42"><a href="#cb12-42" aria-hidden="true" tabindex="-1"></a>ehr <span class="ot">&lt;-</span> <span class="fu">rbind</span>(ehr1, ehr2, ehr3)</span>
<span id="cb12-43"><a href="#cb12-43" aria-hidden="true" tabindex="-1"></a>merged_dt <span class="ot">&lt;-</span> ehr[, .(guid, sex_ehr)][merged_dt, on <span class="ot">=</span> .(<span class="at">guid =</span> guid)]</span>
<span id="cb12-44"><a href="#cb12-44" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(merged_dt,</span>
<span id="cb12-45"><a href="#cb12-45" aria-hidden="true" tabindex="-1"></a>         <span class="at">old =</span> <span class="fu">c</span>(<span class="st">"sex"</span>, <span class="st">"genident"</span>, <span class="st">"sex_ehr"</span>),</span>
<span id="cb12-46"><a href="#cb12-46" aria-hidden="true" tabindex="-1"></a>         <span class="at">new =</span> <span class="fu">c</span>(<span class="st">"REDCap: sex"</span>, <span class="st">"REDCap: genident"</span>, <span class="st">"EHR: sex_ehr"</span>))</span>
<span id="cb12-47"><a href="#cb12-47" aria-hidden="true" tabindex="-1"></a>merged_dt[, rescue <span class="sc">:</span><span class="er">=</span> <span class="fu">ifelse</span>(</span>
<span id="cb12-48"><a href="#cb12-48" aria-hidden="true" tabindex="-1"></a>  (SNPSEX <span class="sc">!=</span> <span class="st">`</span><span class="at">REDCap: genident</span><span class="st">`</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(<span class="st">`</span><span class="at">REDCap: genident</span><span class="st">`</span>) <span class="sc">&amp;</span> <span class="fu">is.na</span>(<span class="st">`</span><span class="at">REDCap: sex</span><span class="st">`</span>)) <span class="sc">|</span></span>
<span id="cb12-49"><a href="#cb12-49" aria-hidden="true" tabindex="-1"></a>    (SNPSEX <span class="sc">!=</span> <span class="st">`</span><span class="at">REDCap: sex</span><span class="st">`</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(<span class="st">`</span><span class="at">REDCap: sex</span><span class="st">`</span>) <span class="sc">&amp;</span> <span class="fu">is.na</span>(<span class="st">`</span><span class="at">REDCap: genident</span><span class="st">`</span>)) <span class="sc">|</span></span>
<span id="cb12-50"><a href="#cb12-50" aria-hidden="true" tabindex="-1"></a>    (SNPSEX <span class="sc">!=</span> <span class="st">`</span><span class="at">REDCap: genident</span><span class="st">`</span> <span class="sc">&amp;</span> SNPSEX <span class="sc">!=</span> <span class="st">`</span><span class="at">REDCap: sex</span><span class="st">`</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(<span class="st">`</span><span class="at">REDCap: genident</span><span class="st">`</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(<span class="st">`</span><span class="at">REDCap: sex</span><span class="st">`</span>)),</span>
<span id="cb12-51"><a href="#cb12-51" aria-hidden="true" tabindex="-1"></a>  <span class="st">"NO"</span>,</span>
<span id="cb12-52"><a href="#cb12-52" aria-hidden="true" tabindex="-1"></a>  <span class="st">"YES"</span></span>
<span id="cb12-53"><a href="#cb12-53" aria-hidden="true" tabindex="-1"></a>)]</span>
<span id="cb12-54"><a href="#cb12-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-55"><a href="#cb12-55" aria-hidden="true" tabindex="-1"></a>merged_dt_no <span class="ot">&lt;-</span> merged_dt[<span class="fu">which</span>(merged_dt<span class="sc">$</span>rescue <span class="sc">==</span> <span class="st">"NO"</span>), ]</span>
<span id="cb12-56"><a href="#cb12-56" aria-hidden="true" tabindex="-1"></a>merged_dt_yes <span class="ot">&lt;-</span> merged_dt[<span class="fu">which</span>(merged_dt<span class="sc">$</span>rescue <span class="sc">==</span> <span class="st">"YES"</span>), ]</span>
<span id="cb12-57"><a href="#cb12-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-58"><a href="#cb12-58" aria-hidden="true" tabindex="-1"></a>drop_sex_discrepancy <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb12-59"><a href="#cb12-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">ID =</span> merged_dt_no<span class="sc">$</span>FID,</span>
<span id="cb12-60"><a href="#cb12-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">IID =</span> merged_dt_no<span class="sc">$</span>IID</span>
<span id="cb12-61"><a href="#cb12-61" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-62"><a href="#cb12-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-63"><a href="#cb12-63" aria-hidden="true" tabindex="-1"></a>related_drop <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"./helpers/A2CPS_related_drop.txt"</span>)</span>
<span id="cb12-64"><a href="#cb12-64" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(related_drop, <span class="fu">c</span>(<span class="st">"ID"</span>, <span class="st">"IID"</span>))</span>
<span id="cb12-65"><a href="#cb12-65" aria-hidden="true" tabindex="-1"></a>drop_sex_discrepancy_and_related <span class="ot">&lt;-</span> <span class="fu">rbind</span>(drop_sex_discrepancy, related_drop)</span>
<span id="cb12-66"><a href="#cb12-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-67"><a href="#cb12-67" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(drop_sex_discrepancy_and_related, <span class="at">col.names =</span> F, <span class="at">sep =</span> <span class="st">" "</span>, <span class="st">"helpers/A2CPS_related_and_sex_discrepancy_drop.txt"</span>)</span>
<span id="cb12-68"><a href="#cb12-68" aria-hidden="true" tabindex="-1"></a>update_sex <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb12-69"><a href="#cb12-69" aria-hidden="true" tabindex="-1"></a>  <span class="at">ID =</span> merged_dt_yes<span class="sc">$</span>FID,</span>
<span id="cb12-70"><a href="#cb12-70" aria-hidden="true" tabindex="-1"></a>  <span class="at">IID =</span> merged_dt_yes<span class="sc">$</span>IID,</span>
<span id="cb12-71"><a href="#cb12-71" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sex =</span> merged_dt_yes<span class="sc">$</span>SNPSEX</span>
<span id="cb12-72"><a href="#cb12-72" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-73"><a href="#cb12-73" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(update_sex, <span class="at">col.names =</span> F, <span class="at">sep =</span> <span class="st">" "</span>, <span class="st">"./helpers/A2CPS_update_sex.txt"</span>)</span>
<span id="cb12-74"><a href="#cb12-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-75"><a href="#cb12-75" aria-hidden="true" tabindex="-1"></a><span class="do">## Graphing</span></span>
<span id="cb12-76"><a href="#cb12-76" aria-hidden="true" tabindex="-1"></a>ok_df<span class="sc">$</span>LegendStatus <span class="ot">&lt;-</span> <span class="st">"OK"</span></span>
<span id="cb12-77"><a href="#cb12-77" aria-hidden="true" tabindex="-1"></a>problem_df<span class="sc">$</span>LegendStatus <span class="ot">&lt;-</span> <span class="st">"PROBLEM"</span></span>
<span id="cb12-78"><a href="#cb12-78" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(ok_df, problem_df)</span>
<span id="cb12-79"><a href="#cb12-79" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"./plots/A2CPS_RL1-5_Sex_Typing.png"</span>, <span class="at">width =</span> <span class="dv">3000</span>, <span class="at">height =</span> <span class="dv">2000</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb12-80"><a href="#cb12-80" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb12-81"><a href="#cb12-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> ok_df, <span class="fu">aes</span>(<span class="at">x =</span> F, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="fu">factor</span>(PEDSEX), <span class="at">shape =</span> <span class="st">"OK"</span>, <span class="at">size =</span> <span class="st">"OK"</span>, <span class="at">alpha =</span> <span class="st">"OK"</span>),</span>
<span id="cb12-82"><a href="#cb12-82" aria-hidden="true" tabindex="-1"></a>               <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="dv">0</span>, <span class="at">height =</span> <span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb12-83"><a href="#cb12-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> problem_df, <span class="fu">aes</span>(<span class="at">x =</span> F, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="fu">factor</span>(PEDSEX), <span class="at">shape =</span> <span class="st">"PROBLEM"</span>, <span class="at">size =</span> <span class="st">"PROBLEM"</span>, <span class="at">alpha =</span> <span class="st">"PROBLEM"</span>),</span>
<span id="cb12-84"><a href="#cb12-84" aria-hidden="true" tabindex="-1"></a>               <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="dv">0</span>, <span class="at">height =</span> <span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb12-85"><a href="#cb12-85" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"black"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"blue"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"magenta"</span>)) <span class="sc">+</span></span>
<span id="cb12-86"><a href="#cb12-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_shape_manual</span>(<span class="at">name =</span> <span class="st">"Status"</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"OK"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"PROBLEM"</span> <span class="ot">=</span> <span class="dv">19</span>)) <span class="sc">+</span></span>
<span id="cb12-87"><a href="#cb12-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_size_manual</span>(<span class="at">name =</span> <span class="st">"Status"</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"OK"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"PROBLEM"</span> <span class="ot">=</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb12-88"><a href="#cb12-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_alpha_manual</span>(<span class="at">name =</span> <span class="st">"Status"</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"OK"</span> <span class="ot">=</span> <span class="fl">0.4</span>, <span class="st">"PROBLEM"</span> <span class="ot">=</span> <span class="fl">0.9</span>)) <span class="sc">+</span></span>
<span id="cb12-89"><a href="#cb12-89" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"A2CPS Sex Discrepancies"</span>,</span>
<span id="cb12-90"><a href="#cb12-90" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"F Statistic"</span>,</span>
<span id="cb12-91"><a href="#cb12-91" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">""</span>,</span>
<span id="cb12-92"><a href="#cb12-92" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">"PEDSEX"</span>) <span class="sc">+</span></span>
<span id="cb12-93"><a href="#cb12-93" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb12-94"><a href="#cb12-94" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"top"</span>,</span>
<span id="cb12-95"><a href="#cb12-95" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb12-96"><a href="#cb12-96" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb12-97"><a href="#cb12-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb12-98"><a href="#cb12-98" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<details>
<summary>
Click to show Supplementary RScript #5: Subsetting just Freeze 2 Participants (Freeze2.R)
</summary>
<div class="sourceCode" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>user_wd <span class="ot">&lt;-</span> args[<span class="dv">1</span>]</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(user_wd)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"./A2CPS_RL1-5_QCed/"</span>)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>current_biospecimen_id <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, <span class="fu">fread</span>(<span class="st">"A2CPS_RL1-5_QCed.fam"</span>)<span class="sc">$</span>V2)</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>freeze2 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"freeze2_ids_ucsd_gv.csv"</span>)<span class="sc">$</span>biospecimen_id</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>drop_not_in_freeze2_list <span class="ot">&lt;-</span> current_biospecimen_id[<span class="fu">which</span>(current_biospecimen_id <span class="sc">%notin%</span> freeze2)]</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>fam.file <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"A2CPS_RL1-5_QCed.fam"</span>)</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>fam.file[, prefix <span class="sc">:</span><span class="er">=</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, V2)]</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>matched_rows <span class="ot">&lt;-</span> fam.file[prefix <span class="sc">%in%</span> drop_not_in_freeze2_list, .(V1, V2)]</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(matched_rows, <span class="at">col.names =</span> F, <span class="at">sep =</span> <span class="st">" "</span>, <span class="st">"notFreeze2.txt"</span>)</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>previous_drops <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/Users/sethberke/a2cps.gv.qc.auto/A2CPS_RL1-5_Preparation_Files/A2CPS_RL1-5_Final_Sample_Drop_List.txt"</span>)</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>total_drops <span class="ot">&lt;-</span> <span class="fu">rbind</span>(matched_rows, previous_drops)</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(total_drops, <span class="at">col.names =</span> F, <span class="at">sep =</span> <span class="st">" "</span>, <span class="st">"/Users/sethberke/a2cps.gv.qc.auto/A2CPS_RL1-5_Preparation_Files/A2CPS_RL1-5_Final_Sample_Drop_List_Only_Freeze_2.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<hr>
</section>
</section>
<section id="principal-component-analysis" class="level3" data-number="1.3.4">
<h3 data-number="1.3.4" class="anchored" data-anchor-id="principal-component-analysis"><span class="header-section-number">1.3.4</span> Principal Component Analysis</h3>
<p>Systematic differences in ancestry between groups – population stratification – can lead to spurious associations with allele frequencies, and so multiethnic analyses often include genetic principal components to account for the variation imposed by population structure. Here we plotted the PCA values across two axes, and can see distinct populations, which must be accounted for in many study designs, such as polygenic risk score (PRS) calculations.</p>
<details>
<summary>
Click to show Principal Components Bash Script, which utilizes Plink 1.9
</summary>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Principal Components Analysis </span><span class="al">###</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> principle_components_analysis</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step a: LD Pruning window of 50kb, step size 10bp, and r^w threshold of 0.1</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> principle_components_analysis/LD_Pruning</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> ./A2CPS_Freeze_2 <span class="at">--double-id</span> <span class="at">--allow-extra-chr</span> <span class="at">--indep-pairwise</span> 50 10 0.1 <span class="at">--out</span> ./principle_components_analysis/LD_Pruning/a2cps_rl12345_LDPruned</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step b: Extract SNPs in equilibrium</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> principle_components_analysis/LD_Pruned</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> ./A2CPS_Freeze_2 <span class="at">--double-id</span> <span class="at">--allow-extra-chr</span> <span class="at">--extract</span> ./principle_components_analysis/LD_Pruning/a2cps_rl12345_LDPruned.prune.in <span class="at">--make-bed</span> <span class="at">--out</span> ./principle_components_analysis/LD_Pruned/a2cps_rl12345_LDPrunedFinal</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step c: Derive PCA Eigenvalues and Eigenvectors</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mkdir</span> principle_components_analysis/PCA</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    <span class="ex">./plink</span> <span class="at">--bfile</span> principle_components_analysis/LD_Pruned/a2cps_rl12345_LDPrunedFinal <span class="at">--double-id</span> <span class="at">--allow-extra-chr</span>  <span class="at">--pca</span> <span class="at">--out</span> principle_components_analysis/PCA/pca_info</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Step d: Graph Scree and PCA Plots</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    <span class="ex">Rscript</span> ./scripts/pca.r <span class="st">"</span><span class="va">$USER_WD</span><span class="st">"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<details>
<summary>
Click to show Supplementary RScript #6: Principal Components Analysis (pca.r)
</summary>
<div class="sourceCode" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load required packages</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httr)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(writexl)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>user_wd <span class="ot">&lt;-</span> args[<span class="dv">1</span>]</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(user_wd)</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>base_url<span class="ot">=</span><span class="st">"https://vbr-api.a2cps.cloud"</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>username <span class="ot">&lt;-</span> VBR_Username</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>password <span class="ot">&lt;-</span> VBR_Password</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>resp <span class="ot">=</span> <span class="fu">POST</span>(<span class="fu">paste</span>(base_url, <span class="st">"/auth/token"</span>, <span class="at">sep=</span><span class="st">""</span>), <span class="at">body =</span> <span class="fu">list</span>(<span class="at">username=</span>username, <span class="at">password=</span>password), <span class="at">encode =</span> <span class="st">"form"</span>)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>access_token <span class="ot">=</span> <span class="fu">content</span>(resp, <span class="at">as=</span><span class="st">"parsed"</span>)<span class="sc">$</span>access_token</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>resp <span class="ot">=</span> <span class="fu">GET</span>(<span class="fu">paste</span>(base_url, <span class="st">"/biospecimens/private?offset=0&amp;limit=0"</span>, <span class="at">sep=</span><span class="st">""</span>), <span class="fu">add_headers</span>(<span class="at">authorization=</span><span class="fu">paste</span>(<span class="st">"Bearer"</span>, access_token)))</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>bio <span class="ot">&lt;-</span> jsonlite<span class="sc">::</span><span class="fu">fromJSON</span>(<span class="fu">content</span>(resp, <span class="at">as =</span> <span class="st">"text"</span>))</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>bioDF <span class="ot">&lt;-</span> <span class="fu">as.data.table</span>(bio)</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>fam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"./Freeze_2/A2CPS_Freeze_2.fam"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>bioID_PLINK <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, fam<span class="sc">$</span>V2)</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>bioDF_Mapping_Plink <span class="ot">&lt;-</span> bioDF[<span class="fu">which</span>(bioDF<span class="sc">$</span>biospecimen_id <span class="sc">%in%</span> bioID_PLINK), ]</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a><span class="co"># BioID and Race</span></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>bioDF_Mapping_Cols <span class="ot">&lt;-</span> bioDF_Mapping_Plink[, <span class="fu">c</span>(<span class="st">"biospecimen_id"</span>,<span class="st">"race"</span>)]</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>fam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"./Freeze_2/A2CPS_Freeze_2.fam"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>fam<span class="sc">$</span>V2 <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, fam<span class="sc">$</span>V2)</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(fam) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Famid"</span>, <span class="st">"biospecimen_id"</span>, <span class="st">"Fatid"</span>, <span class="st">"Motid"</span>, <span class="st">"Gender"</span>, <span class="st">"Phenotype"</span>)</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>fam_with_race <span class="ot">&lt;-</span> <span class="fu">merge</span>(fam, bioDF_Mapping_Cols, <span class="at">by =</span> <span class="st">"biospecimen_id"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"./Freeze_2/principle_components_analysis/PCA"</span>)</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">read_table2</span>(<span class="st">"./pca_info.eigenvec"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>eigenval <span class="ot">&lt;-</span> <span class="fu">scan</span>(<span class="st">"./pca_info.eigenval"</span>)</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>X2 <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, pca<span class="sc">$</span>X2)</span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> pca[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pca)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"biospecimen_id"</span></span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pca)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(pca)] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>,<span class="dv">1</span><span class="sc">:</span>(<span class="fu">ncol</span>(pca)<span class="sc">-</span><span class="dv">1</span>))</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">merge</span>(pca, bioDF_Mapping_Cols, <span class="at">by =</span> <span class="st">"biospecimen_id"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> pca[<span class="sc">!</span><span class="fu">duplicated</span>(pca<span class="sc">$</span>biospecimen_id), ]</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Subsetting for Graph</span></span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>race <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(pca<span class="sc">$</span>biospecimen_id))</span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>race[<span class="fu">grep</span>(<span class="st">"White"</span>, pca<span class="sc">$</span>race)] <span class="ot">&lt;-</span> <span class="st">"White"</span></span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>race[<span class="fu">grep</span>(<span class="st">"African-American"</span>, pca<span class="sc">$</span>race)] <span class="ot">&lt;-</span> <span class="st">"Black or African-American"</span></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>race[<span class="fu">grep</span>(<span class="st">"Asian"</span>, pca<span class="sc">$</span>race)] <span class="ot">&lt;-</span> <span class="st">"Asian"</span></span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>race[<span class="fu">grep</span>(<span class="st">"American Indian or Alaska Native"</span>, pca<span class="sc">$</span>race)] <span class="ot">&lt;-</span> <span class="st">"American Indian or Alaska Native"</span></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>race[<span class="fu">grep</span>(<span class="st">"Unknown"</span>, pca<span class="sc">$</span>race)] <span class="ot">&lt;-</span> <span class="st">"Unknown"</span></span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>race[<span class="fu">grep</span>(<span class="st">"Not Reported"</span>, pca<span class="sc">$</span>race)] <span class="ot">&lt;-</span> <span class="st">"Not Reported"</span></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">as.tibble</span>(<span class="fu">data.frame</span>(pca, race))</span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>pve <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">pve =</span> eigenval<span class="sc">/</span><span class="fu">sum</span>(eigenval)<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a>percent_variance_explained <span class="ot">&lt;-</span> pve<span class="sc">$</span>pve</span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(user_wd)</span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Graphing</span></span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"./Freeze_2/principle_components_analysis/PCA/A2CPS_RL1-5_PCA_Scree_Plot.png"</span>, <span class="at">width =</span> <span class="dv">3000</span>, <span class="at">height =</span> <span class="dv">2000</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>  a <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pve, <span class="fu">aes</span>(PC, pve)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>)</span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>  a <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Percentage Variance Explained"</span>) <span class="sc">+</span> <span class="fu">theme_light</span>()</span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(a)</span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"./Freeze_2/principle_components_analysis/PCA/A2CPS_RL1-5_PCA_Plot.png"</span>, <span class="at">width =</span> <span class="dv">3000</span>, <span class="at">height =</span> <span class="dv">2000</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb15-65"><a href="#cb15-65" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca, <span class="fu">aes</span>(PC1, PC2, <span class="at">col =</span> race)) <span class="sc">+</span></span>
<span id="cb15-66"><a href="#cb15-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">0.075</span>) <span class="sc">+</span></span>
<span id="cb15-67"><a href="#cb15-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb15-68"><a href="#cb15-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span></span>
<span id="cb15-69"><a href="#cb15-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"PCA Genetic Ancestry"</span>) <span class="sc">+</span></span>
<span id="cb15-70"><a href="#cb15-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">1</span>], <span class="dv">3</span>), <span class="st">"%)"</span>)) <span class="sc">+</span> </span>
<span id="cb15-71"><a href="#cb15-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">signif</span>(pve<span class="sc">$</span>pve[<span class="dv">2</span>], <span class="dv">3</span>), <span class="st">"%)"</span>)) <span class="sc">+</span></span>
<span id="cb15-72"><a href="#cb15-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb15-73"><a href="#cb15-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">aspect.ratio =</span> <span class="dv">1</span>,</span>
<span id="cb15-74"><a href="#cb15-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="st">"cm"</span>),</span>
<span id="cb15-75"><a href="#cb15-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">family =</span> <span class="st">"Arial"</span>),</span>
<span id="cb15-76"><a href="#cb15-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">family =</span> <span class="st">"Arial"</span>),</span>
<span id="cb15-77"><a href="#cb15-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">4</span>, <span class="at">family =</span> <span class="st">"Arial"</span>),</span>
<span id="cb15-78"><a href="#cb15-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>, <span class="at">family =</span> <span class="st">"Arial"</span>),</span>
<span id="cb15-79"><a href="#cb15-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb15-80"><a href="#cb15-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb15-81"><a href="#cb15-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>)</span>
<span id="cb15-82"><a href="#cb15-82" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb15-83"><a href="#cb15-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">name =</span> <span class="st">"Ancestry Group"</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"red"</span>, <span class="st">"orange"</span>, <span class="st">"yellow"</span>, <span class="st">"black"</span>, <span class="st">"black"</span>, <span class="st">"purple"</span>))</span>
<span id="cb15-84"><a href="#cb15-84" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(b)</span>
<span id="cb15-85"><a href="#cb15-85" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb15-86"><a href="#cb15-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-87"><a href="#cb15-87" aria-hidden="true" tabindex="-1"></a><span class="co"># PC Values for each participant</span></span>
<span id="cb15-88"><a href="#cb15-88" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> pca[, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">22</span>)]</span>
<span id="cb15-89"><a href="#cb15-89" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(pca, <span class="st">"./Freeze_2/principle_components_analysis/PCA/A2CPS_RL1-5_PCA_Ind_Info.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-pcascree" class="quarto-float quarto-figure quarto-figure-center anchored" alt="A2CPS PCA Scree Plot">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pcascree-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/A2CPS_Freeze2_PCA_Scree_Plot.png" class="img-fluid figure-img" alt="A2CPS PCA Scree Plot" width="500">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pcascree-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.1: A2CPS PCA Scree Plot
</figcaption>
</figure>
</div>
<div id="fig-pcaplot" class="quarto-float quarto-figure quarto-figure-center anchored" alt="A2CPS PCA Plot">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pcaplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/A2CPS_Freeze2_PCA_Plot.png" class="img-fluid figure-img" alt="A2CPS PCA Plot" width="750">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pcaplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.2: A2CPS PCA Plot
</figcaption>
</figure>
</div>
<hr>
</section>
<section id="sex-typing" class="level3" data-number="1.3.5">
<h3 data-number="1.3.5" class="anchored" data-anchor-id="sex-typing"><span class="header-section-number">1.3.5</span> Sex Typing</h3>
<p>This procedure confirms the gender of each participant as documented in the study by testing X Chromosome homozygosity levels. Male participants are haploid for the X chromosome whereas females are diploid, thus we expect a high deviation from the genotypes predicted by Hardy-Weinberg equilibrium (HWE) (0/0, 0/1, 1/1) with male participants, as their alleles should exclusively be either 0 or 1 (hemizygous). This deviation is captured by the F Statistic. As a general rule, an F Stat of 0.3 is the maximum seen for females and 0.75 is the minimum for males (as reflected by the chart below). This chart demonstrates the deviation from HWE as we expect ≈ 1 almost always for males, who will consistently deviate from HWE due to X chromosome hemizygosity, and we expect females to have a normal distribution close to around 0 (as they are heterozygous for the x chromosome). We used this chart to visualize sex discrepancies, and checked these against REDCap and EHR data to settle and recover as many sex discrepancies as possible. Refer to supplementary RScript “sex.typing.r” for the code used to generate this graph. In the graph, the blue dots are males, pink are females, and black are cases where sex was undefined in the .fam file.</p>
<div id="fig-sextyping" class="quarto-float quarto-figure quarto-figure-center anchored" alt="A2CPS Sex Typing">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-sextyping-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/A2CPS_QC_Sex_Typing.png" class="img-fluid figure-img" alt="A2CPS Sex Typing" width="500">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-sextyping-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.3: A2CPS Sex Typing Plot
</figcaption>
</figure>
</div>
<hr>
</section>
<section id="relatedness" class="level3" data-number="1.3.6">
<h3 data-number="1.3.6" class="anchored" data-anchor-id="relatedness"><span class="header-section-number">1.3.6</span> Relatedness</h3>
<p>Allele-sharing percentages, determined during identity-by descent analyses, can reveal relatedness between participants, including the presence of duplicate samples. For example, full siblings are expected to share approximately 50% of their SNPs at a single allele (represented by the Z1 value: e.g., both individuals having “A” in genotypes AA and AB) and about 25% of their SNPs at both alleles (Z2 value: e.g., both having AA at a locus). In contrast, duplicate samples should share nearly 100% of their alleles, resulting in a Z2 value close to 1. To avoid bias in genome-wide association studies due to overrepresentation of related or duplicate individuals, we removed one individual from each identified pair, prioritizing the retention of the sample with higher genotype call quality. In the relatedness plot, sibling pairs appear around Z1 ≈ 0.5, while duplicates cluster in the bottom-left corner (Z0 ≈ 0, Z1 ≈ 0, Z2 ≈ 1). See the supplementary R script het.genome.r for the code used to generate this plot.</p>
<div id="fig-relatedness" class="quarto-float quarto-figure quarto-figure-center anchored" alt="A2CPS Relatedness">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-relatedness-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="images/A2CPS_QC_Relatedness.png" class="img-fluid figure-img" alt="A2CPS Relatedness" width="500">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-relatedness-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.4: A2CPS Relatedness Plot
</figcaption>
</figure>
</div>
<hr>
</section>
<section id="imputation" class="level3" data-number="1.3.7">
<h3 data-number="1.3.7" class="anchored" data-anchor-id="imputation"><span class="header-section-number">1.3.7</span> Imputation</h3>
<p>Imputation is the process of statistically inferring whole genome sequences from SNP Array data using haplotype reference panels. For our imputation we used the TOPMed Imputation Server that implements the TOPMed R3 reference panel. View <a href="genomic_imputation.html" class="quarto-xref"><span>Chapter 2</span></a> for imputation details.</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link" aria-label="Preface">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Preface</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./genomic_imputation.html" class="pagination-link" aria-label="Imputed Genomics">
        <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Imputed Genomics</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>